### DISCUSSION TITLE
Making struct fields required by default

### DISCUSSION THREAD

**Discussion by ClementNerma** [vlang/v#14657] [Ideas]
Hi there!

I've recently started to learn V and I must say it's an amazing language, lots of excellent ideas which makes it very nice to use.

I have however one big problem with the ways structs are defined. V makes a lot of assessments about how it promotes safety in the language (which is indeed true).

But when it comes to struct, it really bothers me that all fields are *optional* by default. It's very easy to make a mistake and forget to specify a field , which will then be set to it's default value. 

This is for me a very big problem as even languages which are not geared as much towards safety such as TypeScript handle this perfectly.

Now I understand it wouldn't be possible to make such a change to the language independently of the fact it's relevant or not, as it would be break many existing scripts.

But would it be possible to have something like a flag or an attribute at the top of the file to indicate the compiler to consider all fields as required by default? This would be incredibly helpful to avoid such errors.

What do you think?

**Comment by JalonSolov** [comment]
V doesn't support marking the whole struct as required, but you can set an attribute on individual fields:  https://github.com/vlang/v/blob/master/doc/docs.md#required-fields

**Comment by ClementNerma** [comment]
> V doesn't support marking the whole struct as required, but you can set an attribute on individual fields: https://github.com/vlang/v/blob/master/doc/docs.md#required-fields

Which is exactly the problem I have: I don't want to specify it every time as it makes the code less readable, and it's error prone as it can be forgotten easily on a new struct for instance.

So what I'd like is to have a flag or an attribute to make *all* struct fields required.

**Comment by ghost** [comment]
I think such an attribute would be nice to have.
Someone please make one.
